aten/src/TH/generic/THTensorMoreMath.cpp

```diff
@@ -28,9 +28,18 @@ int THTensor_(equal)(THTensor *ta, THTensor* tb)
     scalar_t *tbp = tb->data<scalar_t>();
     ptrdiff_t sz = THTensor_(nElement)(ta);
     ptrdiff_t i;
-    for (i=0; i<sz; ++i){
-      if(tap[i] != tbp[i]) return 0;
-    }
+    at::parallel_for(
+        0,
+        sz,
+        HYPER_TH_OMP_OVERHEAD_THRESHOLD,
```

Comment:
I *think* HYPER_TH_OMP_OVERHEAD_THRESHOLD is used for more expensive functions like cosh so it can generate too fine-grain tasks. Maybe TH_OMP_OVERHEAD_THRESHOLD is a right one to use here?

Reply:
I'd suggest running a simple benchmark to tune this param

Reply:
The definition of `HYPER_TH_OMP_OVERHEAD_THRESHOLD` and `TH_OMP_OVERHEAD_THRESHOLD` can be found in https://github.com/pytorch/pytorch/blob/master/aten/src/TH/generic/THTensorApply.hpp#L7-L10 .
As you mentioned,  `HYPER_TH_OMP_OVERHEAD_THRESHOLD` is used for more fine-grained tasks since HYPER_TH_OMP_OVERHEAD_THRESHOLD = TH_OMP_OVERHEAD_THRESHOLD / 16.

However, as I measured, the performance differences between these different grain size are subtle:

(grain size is the third parameter when using `at::parallel_for`. As measured below, the performance differences between these different grain size seem to be subtle.)

- grain size = `TH_OMP_OVERHEAD_THRESHOLD`:
```
equal                    1.43%            36.056ms         1.43%            36.056ms         500.783us        NaN              0.000us          0.000us          72               [[61, 64, 1024], [61, 64, 1024]]
```

- grain size = `HYPER_TH_OMP_OVERHEAD_THRESHOLD`:
```
equal                    1.41%            35.126ms         1.41%            35.126ms         487.855us        NaN              0.000us          0.000us          72               [[61, 64, 1024], [61, 64, 1024]]
```

- grain size = 1:
```
equal                    1.43%            35.632ms         1.43%            35.632ms         494.886us        NaN              0.000us          0.000us          72               [[61, 64, 1024], [61, 64, 1024]]
```